python/cugraph/cugraph/structure/replication.py

```diff
@@ -0,0 +1,350 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dask_cudf
+import cudf
+from dask.distributed import wait, default_client
+import numpy as np
+from pylibcugraph import (
+    ResourceHandle,
+    replication as pylibcugraph_replication,
+)
+
+from cugraph.dask.common.part_utils import (
+    get_persisted_df_worker_map,
+    persist_dask_df_equal_parts_per_worker,
+)
+
+import dask
+import cupy as cp
+import cugraph.dask.comms.comms as Comms
+from typing import Union, Tuple
+
+
+def convert_to_cudf(cp_arrays: Tuple[cp.ndarray], col_names: list) -> cudf.DataFrame:
```

Comment:
If this is only used in this file then it might be better to name it `_convert_to_cudf`, since otherwise it seems like it's supposed to be a general-purpose utility, and maybe we'd put it in a common place for utils.

Reply:
Right! several MG algos use this pattern and it should definitely be a general-purpose function. I used the `_` for now and added a FIXME as this will require refactoring a lot of our algos